yolo export: AssertionError in ultralytics/engine/exporter.py:export_coreml

Automated fuzzing found a reproducible failure (confirmed 2/2 runs).

### Reproduce

```bash
yolo export classify model=yolo26n-cls.pt imgsz=32 format=coreml opset=12 batch=2
```

### Details

- Outcome: `bug-candidate` · strategy: `combo` · task: `classify` · seed: `29713810438`
- Run: https://github.com/ultralytics/ultralytics/actions/runs/29713810438

### Traceback (tail)

```
Traceback (most recent call last):
  File "<frozen runpy>", line 203, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/runner/work/ultralytics/ultralytics/ultralytics/cfg/__init__.py", line 1146, in <module>
    entrypoint(debug="")
    ~~~~~~~~~~^^^^^^^^^^
  File "/Users/runner/work/ultralytics/ultralytics/ultralytics/cfg/__init__.py", line 1105, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/runner/work/ultralytics/ultralytics/ultralytics/engine/model.py", line 722, in export
    return Exporter(overrides=args, _callbacks=self.callbacks)(model=self.model)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/ultralytics/ultralytics/ultralytics/engine/exporter.py", line 891, in __call__
    f = getattr(self, f"export_{fmt}")()
  File "/Users/runner/work/ultralytics/ultralytics/ultralytics/engine/exporter.py", line 468, in outer_func
    raise e
  File "/Users/runner/work/ultralytics/ultralytics/ultralytics/engine/exporter.py", line 454, in outer_func
    f = inner_func(*args, **kwargs)  # exported file/dir or tuple of (file/dir, *)
  File "/Users/runner/work/ultralytics/ultralytics/ultralytics/engine/exporter.py", line 1210, in export_coreml
    assert self.args.dynamic, (
           ^^^^^^^^^^^^^^^^^
AssertionError: batch sizes > 1 are not supported without 'dynamic=True' for CoreML export. Please retry at 'dynamic=True'.
```

### Environment

```
OS: macOS-26.4-arm64-arm-64bit-Mach-O
Environment: Darwin
Python: 3.13.14
Install: git
Path: /Users/runner/work/ultralytics/ultralytics/ultralytics
RAM: 7.00 GB
Disk: 225.5/319.5 GB
CPU: Apple M1 (Virtual)
CPU count: 3
GPU: None
GPU count: None
CUDA: None
Package Info: {'filelock': '✅ 3.31.1>=3.16.1', 'numpy': '✅ 2.2.6>=1.23.0', 'matplotlib': '✅ 3.11.1>=3.3.0', 'opencv-python': '✅ 5.0.0.93!=4.13.0.90,>=4.6.0', 'pillow': '✅ 12.3.0>=7.1.2', 'pyyaml': '✅ 6.0.3>=5.3.1', 'requests': '✅ 2.34.2>=2.23.0', 'torch': '✅ 2.13.0!=2.4.0,>=1.8.0; sys_platform == "win32"', 'torchvision': '✅ 0.28.0>=0.9.0', 'psutil': '✅ 7.2.2>=5.8.0', 'polars': '✅ 1.42.1>=0.20.0', 'nvidia-ml-py': '✅ 13.610.43>=12.0.0', 'ultralytics-thop': '✅ 2.0.20>=2.0.18'}
GitHub Info: {'RUNNER_OS': 'macOS', 'GITHUB_EVENT_NAME': 'schedule', 'GITHUB_WORKFLOW': 'Fuzz', 'GITHUB_ACTOR': 'glenn-jocher', 'GITHUB_REPOSITORY': 'ultralytics/ultralytics', 'GITHUB_REPOSITORY_OWNER': 'ultralytics'}
```

<!-- fuzz-signature: 904922b5df10 -->

--- a/docs/en/guides/ros-quickstart.md
+++ b/docs/en/guides/ros-quickstart.md
@@ -90,6 +90,7 @@ class UltralyticsNode(Node):
     """Run YOLO detection on ROS2 image messages."""
 
     def __init__(self):
+        """Initialize the ROS2 node, model, and image interfaces."""
         super().__init__("ultralytics")
         self.bridge = cv_bridge.CvBridge()
         self.model = YOLO("yolo26m.pt")

--- a/ultralytics/engine/exporter.py
+++ b/ultralytics/engine/exporter.py
@@ -729,6 +729,18 @@ class Exporter:
                 raise ValueError("Alibaba MNN export does not support combining 'dynamic=True' with 'nms=True'.")
             if model.task not in {"detect", "pose"}:
                 raise ValueError("Alibaba MNN export with 'nms=True' only supports detect and pose models.")
+        if fmt == "coreml":
+            if self.args.batch > 1:
+                assert self.args.dynamic, (
+                    "batch sizes > 1 are not supported without 'dynamic=True' for CoreML export. Please retry at 'dynamic=True'."
+                )
+            if self.args.dynamic:
+                assert not self.args.nms, (
+                    "'nms=True' cannot be used together with 'dynamic=True' for CoreML export. Please disable one of them."
+                )
+                assert model.task != "classify" and not isinstance(model.model[-1], RTDETRDecoder), (
+                    "'dynamic=True' is not supported for CoreML classification or RT-DETR models."
+                )
         if (fmt in {"engine", "coreml"} or self.args.nms) and self.args.dynamic and self.args.batch == 1:
             LOGGER.warning(
                 f"'dynamic=True' model with '{'nms=True' if self.args.nms else f'format={self.args.format}'}' requires max batch size, i.e. 'batch=16'"
@@ -1204,15 +1216,6 @@ class Exporter:
 
         assert not WINDOWS, "CoreML export is not supported on Windows, please run on macOS or Linux."
         assert TORCH_1_11, "CoreML export requires torch>=1.11"
-        if self.args.batch > 1:
-            assert self.args.dynamic, (
-                "batch sizes > 1 are not supported without 'dynamic=True' for CoreML export. Please retry at 'dynamic=True'."
-            )
-        if self.args.dynamic:
-            assert not self.args.nms, (
-                "'nms=True' cannot be used together with 'dynamic=True' for CoreML export. Please disable one of them."
-            )
-            assert self.model.task != "classify", "'dynamic=True' is not supported for CoreML classification models."
         f = self.file.with_suffix(".mlmodel" if mlmodel else ".mlpackage")
         if f.is_dir():
             shutil.rmtree(f)

